Products/ZenModel/migrate/convertRunsToCommands.py

```diff
@@ -0,0 +1,125 @@
+"""
+import logging
+log = logging.getLogger("zen.migrate")
+
+import Migrate
+import servicemigration as sm
+sm.require("1.0.0")
+
+
+class ConvertRunsToCommands(Migrate.Step):
+
+    version = Migrate.Version(5, 0, 70)
+
+    def cutover(self, dmd):
+        try:
+            ctx = sm.ServiceContext()
+        except sm.ServiceMigrationError:
+            log.info("Couldn't generate service context, skipping.")
+            return
+
+        # Update the zope service commands.
+        zopes = filter(lambda s: s.name == "Zope", ctx.services)
+        if len(zopes) == 1:
+            temp = commandListDict(zopes[0].commands)
+            temp.update(zopeCommands)
+            zopes[0].commands = commandDictList(temp)
+        mariadbModels = filter(lambda s: s.name == "mariadb-model", ctx.services)
+        if len(mariadbModels) == 1:
+            temp = commandListDict(mariadbModels[0].commands)
+            temp.update(mariadbModelCommands)
+            mariadbModels[0].commands = commandDictList(temp)
+        mariadbs = filter(lambda s: s.name == "mariadb", ctx.services)
+        if len(mariadbs) == 1:
+            temp = commandListDict(mariadbs[0].commands)
+            temp.update(mariadbCommands)
+            mariadbs[0].commands = commandDictList(temp)
+        # Commit our changes.
+        ctx.commit()
+
+
+ConvertRunsToCommands()
+
+
+def commandListDict(commandList):
+    commandDict = {}
+    for command in commandList:
+        commandDict[command.name] = {
+            "Command": command.command,
+            "CommitOnSuccess": command.commitOnSuccess
+        }
+    return commandDict
+
+
+def commandDictList(commandDict):
+    commandList = []
+    for k, v in commandDict.iteritems():
+        commandList.append(sm.Command(k, v["Command"], v["CommitOnSuccess"]))
+    return commandList
+
+
+zopeCommands = {
+    "zendmd": {
+        "Command": "${ZENHOME:-/opt/zenoss}/bin/zenrun zendmd",
+        "CommitOnSuccess": True
+    },
+    "upgrade": {
+        "Command": "${ZENHOME:-/opt/zenoss}/bin/zenrun upgrade.sh doUpgrade",
+        "CommitOnSuccess": False
+    },
+    "apply-custom-patches": {
+        "Command": "${ZENHOME:-/opt/zenoss}/bin/zenrun patch.sh applyPatches",
+        "CommitOnSuccess": True
+    },
+    "reportmail": {
+        "Command": "${ZENHOME:-/opt/zenoss}/bin/zenrun reportmail.sh",
+        "CommitOnSuccess": False
+    },
+    "help": {
+        "Command": "${ZENHOME:-/opt/zenoss}/bin/zenrun help.sh",
+        "CommitOnSuccess": False
+    },
+    "zenpack-manager": {
+        "Command": "${ZENHOME:-/opt/zenoss}/bin/zenrun zenpack-manager.sh --service-id={{.ID}}",
+        "CommitOnSuccess": True
+    },
+    "zenpack": {
+        "Command": "${ZENHOME:-/opt/zenoss}/bin/zenrun zenpack.sh --service-id={{.ID}}",
+        "CommitOnSuccess": False
+    },
+    "install-percona": {
+        "Command": "${ZENHOME:-/opt/zenoss}/bin/zenrun percona.sh install",
+        "CommitOnSuccess": True
```

Comment:
Previously, the Runs for `zendmd` and `install-percona` specified `NOCOMMIT=true`, so for backwards-compatibility, I would expect the replacement Commands to use `CommitOnSuccess=False`.  So either, I'm misinterpreting a double-negative somewhere, or the old definition was wrong and needs to be changed (in which case this PR is correct), or these definitions are not backwards compatible.